dipy/data/fetcher.py

```diff
@@ -1628,8 +1635,141 @@ def read_five_af_bundles():
 
         bundles = []
         for sub in subjects:
-            fname = os.path.join(temp_dir, sub, 'AF_L.trk')
+            fname = pjoin(temp_dir, sub, 'AF_L.trk')
             bundle_obj = load_trk(fname, 'same', bbox_valid_check=False)
             bundles.append(bundle_obj.streamlines)
 
     return bundles
+
+
+def to_bids_description(path, fname='dataset_description.json',
+                        BIDSVersion="1.4.0", **kwargs):
+    """Dumps a dict into a bids description at the given location"""
+    kwargs.update({"BIDSVersion": BIDSVersion})
+    desc_file = op.join(path, fname)
+    with open(desc_file, 'w') as outfile:
+        json.dump(kwargs, outfile)
+
+
+def fetch_hcp(subjects,
+              hcp_bucket='hcp-openaccess',
+              profile_name="hcp",
+              path=None,
+              study='HCP_1200',
+              aws_access_key_id=None,
+              aws_secret_access_key=None):
+    """
+    Fetch HCP diffusion data and arrange it in a manner that resembles the
+    BIDS [1]_ specification.
+
+    Parameters
+    ----------
+    subjects : list
+        Each item is an integer, identifying one of the HCP subjects
```

Comment:
can you add a example here, I made the stupid mistake to just add an int instead of a list[int]

Reply:
Instead of adding the example, I added a bit of code to recover from the case where users provide just one subject (either as integer or as string), because I think that's a typical failure mode.